src/server/response.cpp

```diff
@@ -203,9 +203,9 @@ void ContentResponse::introduce_taskbar()
   data.set("title", m_bookTitle);
   data.set("withlibrarybutton", m_withLibraryButton);
   auto head_content = render_template(RESOURCE::templates::head_part_html, data);
-  m_content = appendToFirstOccurence(
+  m_content = prependToFirstOccurence(
     m_content,
-    "<head[^>]*>",
+    "</head[^>]*>",
```

Comment:
For a closing tag we don't need the **`[^>]*`** in the regular expression (or, at most, we can allow some whitespace there).

Reply:
Also I think that for this particular case the strings don't need to go through the UTF-8 conversion in both directions, i.e. the regular STL functions would work perfectly fine, but that comment probably belongs to the premature optimization category.

Reply:
@veloman-yunkan I have kept `[^>]*` as I thought this is just the best approach to handle comments/tabs/spaces. I still think so? Regarding the second part, I have really not much of a clue. Because of that, I would tend to keep it. But I can remove it if you want.

Reply:
> I have kept `[^>]*` as I thought this is just the best approach to handle tabs/spaces/.....

That's definitely not the best approach to handle whitespace. That patter will also match `</header>`, `</headache>`. Better explicitly use a whitespace designator. And probably you would also want to allow whitespace *before* `head` too?

Reply:
@veloman-yunkan Hopefully better now?